### ISSUE TITLE
Error when running service 'Important file(s) for '/etc/nginx/conf.d/default.conf' are missing or empty'

### ISSUE THREAD

**Issue by andreiyusupau** [JonasAlfredsson/docker-nginx-certbot#251]
Hi, I'm having some troubles running the docker-nginx-certbot container. It seems that some misconfiguration occurred. Is it ok that service writes in log that it `[error] Important file(s) for '/etc/nginx/conf.d/default.conf' are missing or empty, disabling...'`, how could I check where does the error come from?
Thanks!

docker-compose.yml:
```
services:
  proxy:
    image: jonasal/nginx-certbot:5.2.0-nginx1.26.0-alpine
    ports:
      - 80:80
      - 443:443
    restart: unless-stopped
    environment:
      - CERTBOT_EMAIL=someemail@gmail.com
    volumes:
      - ./nginx/conf/ssl/:/etc/nginx/user_conf.d
```

default.conf
```
limit_req_zone $binary_remote_addr zone=one:10m rate=50r/s;

server {
     listen 80;
     listen [::]:80;

     server_name example.com www.example.com;

     server_tokens off;

     location /.well-known/acme-challenge/ {
         root /var/www/certbot;
     }

     location / {
         return 301 https://$host$request_uri;
     }
}

server {
    listen 443 default_server ssl;
    listen [::]:443 ssl;

    server_name example.com;

    ssl_certificate /etc/nginx/ssl/live/example.com/fullchain.pem;
    ssl_certificate_key /etc/nginx/ssl/live/example.com/privkey.pem;
    ssl_session_cache shared:SSL:10m;
    ssl_session_timeout 5m;
    ssl_protocols TLSv1.1 TLSv1.2;
    ssl_prefer_server_ciphers on;
    ssl_dhparam /etc/ssl/certs/dhparam.pem;
    ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA:ECDHE-ECDSA-AES128-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES128-SHA:ECDHE-RSA-AES256-SHA:ECDHE-RSA-AES128-SHA256:ECDHE-RSA-AES256-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES256-SHA256;
    add_header Strict-Transport-Security "max-age=31536000; includeSubdomains";

    location /api/v1 {
        proxy_pass http://example-backend:8080;
        limit_req zone=one burst=20 nodelay;
    }

    location / {
        auth_basic "Restricted";
        auth_basic_user_file  /etc/nginx/.htpasswd;
        proxy_pass http://example-frontend:3000;
        limit_req zone=one burst=20 nodelay;
    }
}

```

logs from container:
```
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
10-listen-on-ipv6-by-default.sh: info: /etc/nginx/conf.d/default.conf is not a file or does not exist
/docker-entrypoint.sh: Sourcing /docker-entrypoint.d/15-local-resolvers.envsh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
/docker-entrypoint.sh: Configuration complete; ready for start up
2024/05/27 21:11:27 [info] Creating symlink '/etc/nginx/conf.d/default.conf' to '/etc/nginx/user_conf.d/default.conf'
2024/05/27 21:11:27 [warning] Could not find non-zero size keyfile file '/etc/nginx/ssl/live/example.com/privkey.pem' in '/etc/nginx/conf.d/default.conf'
2024/05/27 21:11:27 [warning] Could not find non-zero size fullchain file '/etc/nginx/ssl/live/example.com/fullchain.pem' in '/etc/nginx/conf.d/default.conf'
2024/05/27 21:11:27 [warning] Could not find non-zero size dhparam file '/etc/ssl/certs/dhparam.pem' in '/etc/nginx/conf.d/default.conf'
2024/05/27 21:11:27 [error] Important file(s) for '/etc/nginx/conf.d/default.conf' are missing or empty, disabling...
2024/05/27 21:11:27 [info] Starting the Nginx service
2024/05/27 21:11:27 [info] Running the autorenewal service
2024/05/27 21:11:27 [notice] 87#87: using the "epoll" event method
2024/05/27 21:11:27 [notice] 87#87: nginx/1.26.0
2024/05/27 21:11:27 [notice] 87#87: built by gcc 13.2.1 20231014 (Alpine 13.2.1_git20231014)
2024/05/27 21:11:27 [notice] 87#87: OS: Linux 6.5.0-9-generic
2024/05/27 21:11:27 [notice] 87#87: getrlimit(RLIMIT_NOFILE): 1048576:1048576
2024/05/27 21:11:27 [notice] 87#87: start worker processes
2024/05/27 21:11:27 [notice] 87#87: start worker process 93
2024/05/27 21:11:27 [notice] 87#87: start worker process 94
2024/05/27 21:11:27 [warning] Couldn't find the dhparam file '/etc/ssl/certs/dhparam.pem'; creating it...
2024/05/27 21:11:27 [info]
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %                        ATTENTION!                       %
    %                                                         %
    % This script will now create a 2048 bit Diffie-Hellman   %
    % parameter to use during the SSL handshake.              %
    %                                                         %
    % >>>>>      This MIGHT take a VERY long time!      <<<<< %
    %    (Took 65 minutes for 4096 bit on an old 3GHz CPU)    %
    %                                                         %
    % However, there is some randomness involved so it might  %
    % be both faster or slower for you. 2048 is secure enough %
    % for today and quite fast to generate. These files will  %
    % only have to be created once so please be patient.      %
    % A message will be displayed when this process finishes. %
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

2024/05/27 21:11:27 [info] Will now output to the following file: '/etc/ssl/certs/dhparam.pem'
WARNING: can't open config file: /etc/ssl/libressl.cnf
Generating DH parameters, 2048 bit long safe prime, generator 2
This is going to take a long time
.............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................*
2024/05/27 21:12:20 [info]
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    % >>>>>   Diffie-Hellman parameter creation done!   <<<<< %
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

2024/05/27 21:12:20 [info] Starting certificate renewal process
2024/05/27 21:12:20 [warning] Could not find non-zero size keyfile file '/etc/nginx/ssl/live/example.com/privkey.pem' in '/etc/nginx/conf.d/default.conf.nokey'
2024/05/27 21:12:20 [warning] Could not find non-zero size fullchain file '/etc/nginx/ssl/live/example.com/fullchain.pem' in '/etc/nginx/conf.d/default.conf.nokey'
nginx: the configuration file /etc/nginx/nginx.conf syntax is ok
nginx: configuration file /etc/nginx/nginx.conf test is successful
2024/05/27 21:12:20 [notice] 187#187: signal process started
2024/05/27 21:12:20 [notice] 87#87: signal 1 (SIGHUP) received from 187, reconfiguring
2024/05/27 21:12:20 [notice] 87#87: reconfiguring
2024/05/27 21:12:20 [notice] 87#87: using the "epoll" event method
2024/05/27 21:12:20 [notice] 87#87: start worker processes
2024/05/27 21:12:20 [notice] 87#87: start worker process 188
2024/05/27 21:12:20 [notice] 87#87: start worker process 189
2024/05/27 21:12:20 [info] Autorenewal service will now sleep 8d
2024/05/27 21:12:20 [notice] 93#93: gracefully shutting down
2024/05/27 21:12:20 [notice] 94#94: gracefully shutting down
2024/05/27 21:12:20 [notice] 94#94: exiting
2024/05/27 21:12:20 [notice] 93#93: exiting
2024/05/27 21:12:20 [notice] 93#93: exit
2024/05/27 21:12:20 [notice] 94#94: exit
2024/05/27 21:12:20 [notice] 87#87: signal 17 (SIGCHLD) received from 94
2024/05/27 21:12:20 [notice] 87#87: worker process 94 exited with code 0
2024/05/27 21:12:20 [notice] 87#87: signal 29 (SIGIO) received
2024/05/27 21:12:20 [notice] 87#87: signal 17 (SIGCHLD) received from 93
2024/05/27 21:12:20 [notice] 87#87: worker process 93 exited with code 0
2024/05/27 21:12:20 [notice] 87#87: signal 29 (SIGIO) received

```

**Comment by JonasAlfredsson** [comment]
Hi  andreiyusupau, 

You can probably search for the error string here in github to see where in the code it is printed. 

However, I suspect you are making the same mistake as [here](https://github.com/JonasAlfredsson/docker-nginx-certbot/issues/197), so just remove the server listening to port 80 from your config and let [redirector.conf](https://github.com/JonasAlfredsson/docker-nginx-certbot/blob/master/src/nginx_conf.d/redirector.conf) do its work.

**Comment by andreiyusupau** [comment]
Thanks @JonasAlfredsson , everything works perfect!
I've removed the server listening to port 80,
but I think generally the problem was in wrong paths to certificates, so I've updated them to:
```
    ssl_certificate /etc/letsencrypt/live/example.com/fullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/example.com/privkey.pem;
    ssl_trusted_certificate /etc/letsencrypt/live/example.com/chain.pem;
```

P.S. The server listening to port 80 was required to redirect traffic from `http` to `https`, but also to redirect from `www.example.com` to `example.com`. What is the correct way to achieve this, should I redeclare `redirector.conf` or maybe just request certificates for both `www.example.com` and `example.com`?
